pandas/core/generic.py

```diff
@@ -699,18 +699,100 @@ def pop(self, item):
 
     def squeeze(self, axis=None):
         """
-        Squeeze length 1 dimensions.
+        Squeeze 1 dimensional axis objects into scalars.
+
+        If the given axis consists of one dimensional objects, they are turned
+        into scalars. In case no axis is specified, all axes are subject to
+        squeezing. In any case, objects in axes that can't be squeezed are
+        left unchanged.
 
         Parameters
         ----------
-        axis : None, integer or string axis name, optional
-            The axis to squeeze if 1-sized.
+        axis : integer or string, optional
+            A specific axis to squeeze.
 
             .. versionadded:: 0.20.0
 
         Returns
         -------
-        scalar if 1-sized, else original object
+        DataFrame, Series or scalar
+            The projection after squeezing axis.
+
+        See Also
+        --------
+        Series.iloc : Integer-location based indexing for selecting scalars
+        DataFrame.iloc : Integer-location based indexing for selecting Series
```

Comment:
Does it make sense to add `DataFrame.to_series`? I think they do the same in the case of 1 column DataFrame, right?

Reply:
It is somewhat, I think squeezing is most useful in slicing scenarios but perhaps someone might find that a direct conversion is what they really wanted. Will add it too.

Reply:
I don't think `DataFrame.to_series` exists?

Reply:
hehe, that's a good reason to not add it... ;) not sure what I was thinking about, I think I got confused with `Index.to_series`, sorry